### ISSUE TITLE
Should matches be played less frequently?

### ISSUE THREAD

**Issue by arnaudgo** [gcp/leela-zero#800]
Hello,

Very often, there's no new net getting promoted before several multiples of 400 match games are played.
Resources dedicated to those games are then not used for self-games.

A question that comes to my mind: does LZ learn from match games too?

Otherwise, wouldn't it make sense to play matches less often?

Arnaud

**Comment by jkiliani** [comment]
I disagree. We're currently expending 5-10% of resources to matches. If we had a lot less matches, it would in all likelihood take considerably longer to find stronger nets (and the point is to progress, not to just play more games). We have to test so many since until today, there is no reliable prediction at which training step count a particular window will yield a strong net, so there is no alternative to testing many.

**Comment by gcp** [comment]
No.

**Comment by marcocalignano** [comment]
Actually if you change the termination rule you could generate a new network when the MSE is below a certain value, no matter how many steps you need. You could have then three different MSE thresholds and make matches only with these three networks.

**Comment by arnaudgo** [comment]
Okay, thanks all for your comments :-)

Arnaud

**Comment by jkiliani** [comment]
@marcocalignano Sounds interesting, but is there a guarantee that MSE will ever drop below a certain pre-set threshold? There would have to be some sort of termination condition that ends a training run regardless. Also, why not make use of the new test-validation split? Should the Tensorboard plots now not show when test and validation MSE diverge, i.e. overfitting is starting? I tried to get Tensorflow working on my machine but unfortunately didn't manage to, so I can't look into the training loss charts...

**Comment by marcocalignano** [comment]
@jkiliani of course you need a MAX_STEPS number where you stop training the network and just start training another one. This number could be 128K or 256K. The advantage of this is in the current method if 
a network with 64K steps becomes the best network is very likely the the next network with 96K steps and 128K do not become the next best networks. If you look at MSE, can happen that the network stops ad 104K, becomes the best network but with a grater improvement (more ELO). So this method could bring less matches and more improvement. 
Of course you have to look at the MSE of the validation sample to avoid overfitting.

**Comment by marcocalignano** [comment]
@jkiliani It is also not that difficult to implement at  [Line 186](https://github.com/gcp/leela-zero/blob/47ddf38327b80ec5f2303344d1d9c65e25f5a3fb/training/tf/tfprocess.py#L186) you just write 
`if avg_mse_loss < 0.10` instead of the other if.
